pkg/clusteragent/externalmetrics/autoscaler_watcher.go

```diff
@@ -76,19 +90,30 @@ func NewAutoscalerWatcher(refreshPeriod int64, autogenEnabled bool, autogenExpir
 	}
 
 	// Setup HPA
-	var autoscalerLister autoscaler_lister.HorizontalPodAutoscalerLister
+	var autoscalerLister cache.GenericLister
 	var autoscalerListerSynced cache.InformerSynced
 	if informer != nil {
-		autoscalerLister = informer.Autoscaling().V2beta1().HorizontalPodAutoscalers().Lister()
-		autoscalerListerSynced = informer.Autoscaling().V2beta1().HorizontalPodAutoscalers().Informer().HasSynced
+		hpaGVR, err := autoscalers.DiscoverHPAGroupVersionResource(client)
+		if err != nil {
+			log.Errorf("unable to discover HPA GroupVersionResource: %s", err)
```

Comment:
Shouldn't we fallback to a default version here?

Reply:
We... could? Choosing one is complicated though. `v2`?

Reply:
I'd say v2 yes but @DataDog/container-ecosystems should be involved in this decision for sure.
The other option is to return the error and not run the external metric provider.

Reply:
I think returning an error is less surprising behavior, and easier to detect. I'll go with that approach 👍